src/builders/indexable-hierarchy-builder.php

```diff
@@ -178,7 +179,7 @@ private function find_primary_term_id_for_post( $post ) {
 
 		$primary_term = $this->primary_term_repository->find_by_post_id_and_taxonomy( $post->ID, $main_taxonomy, false );
 
-		if ( $primary_term ) {
+		if ( $primary_term && \get_term( $primary_term->term_id ) ) {
```

Comment:
Please add a check here using `is_wp_error` to also account for `WP_Error` returns instead of just `null` returns.